app/controllers/CyaUpdateRecordController.scala

```diff
@@ -66,62 +66,59 @@ class CyaUpdateRecordController @Inject() (
           )
           Ok(view(list, onSubmitAction))
-        case Left(errors)                                => logErrorsAndContinue(errors, recordId)
+        case Left(errors)            =>
+          logErrorsAndContinue(errors, routes.CyaUpdateRecordController.onPageLoadGoodsDescription(recordId))
       }
     }
 
   def onPageLoadTraderReference(recordId: String): Action[AnyContent] =
     (identify andThen getData andThen requireData) { implicit request =>
-      UpdateGoodsRecord.buildTraderReference(request.userAnswers, request.eori, recordId) match {
-        case Right(updateGoodsRecord: UpdateGoodsRecord) =>
+      UpdateGoodsRecord.buildTraderReference(request.userAnswers, recordId) match {
+        case Right(traderReference) =>
           val onSubmitAction = routes.CyaUpdateRecordController.onSubmitTraderReference(recordId)
 
           val list = SummaryListViewModel(
-            Seq(
-              //TODO remove .get
-              TraderReferenceSummary.row(updateGoodsRecord.traderReference.get, recordId, CheckMode)
-            )
+            Seq(TraderReferenceSummary.row(traderReference, recordId, CheckMode))
           )
           Ok(view(list, onSubmitAction))
-        case Left(errors)                                => logErrorsAndContinue(errors, recordId)
+        case Left(errors)           =>
+          logErrorsAndContinue(errors, routes.CyaUpdateRecordController.onPageLoadTraderReference(recordId))
       }
     }
 
   def onPageLoadCommodityCode(recordId: String): Action[AnyContent] =
     (identify andThen getData andThen requireData) { implicit request =>
-      UpdateGoodsRecord.buildCommodityCode(request.userAnswers, request.eori, recordId) match {
-        case Right(updateGoodsRecord: UpdateGoodsRecord) =>
+      UpdateGoodsRecord.buildCommodityCode(request.userAnswers, recordId) match {
+        case Right(commodity) =>
           val onSubmitAction = routes.CyaUpdateRecordController.onSubmitCommodityCode(recordId)
 
           val list = SummaryListViewModel(
-            Seq(
-              //TODO remove .get
-              CommodityCodeSummary.row(updateGoodsRecord.commodityCode.map(_.commodityCode).get, recordId, CheckMode)
-            )
+            Seq(CommodityCodeSummary.row(commodity.commodityCode, recordId, CheckMode))
           )
           Ok(view(list, onSubmitAction))
-        case Left(errors)                                => logErrorsAndContinue(errors, recordId)
+        case Left(errors)     =>
+          logErrorsAndContinue(errors, routes.CyaUpdateRecordController.onPageLoadCommodityCode(recordId))
       }
     }
 
-  def getCountryOfOriginAnswer(userAnswers: UserAnswers, recordId: String)(implicit
+  private def getCountryOfOriginAnswer(userAnswers: UserAnswers, recordId: String)(implicit
     request: Request[_]
   ): Future[String] =
     userAnswers.get(CountryOfOriginUpdatePage(recordId)) match {
```

Comment:
do we need to check what happens if it doesn't have an answer?

Reply:
Not needed